packages/eslint-plugin/src/rules/call-super-on-override.ts

```diff
@@ -0,0 +1,119 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+
+import * as utils from '../util';
+
+type MessageIds = 'missingSuperMethodCall';
+
+export default utils.createRule<[], MessageIds>({
+  name: 'call-super-on-override',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Require overridden methods to call super.method in their body',
+      recommended: false,
+      requiresTypeChecking: false,
+    },
+    messages: {
+      missingSuperMethodCall:
+        "Use 'super{{property}}{{parameterTuple}}' to avoid missing super class method implementations",
+    },
+    fixable: 'code',
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          topLevel: {
+            type: 'boolean',
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      'MethodDefinition[override=true][kind="method"]'(
+        node: TSESTree.MethodDefinition,
+      ): void {
+        let methodName = '',
+          methodNameIsLiteral = false,
+          methodNameIsNull = false; // don't add quotes for error message on [null] case
+
+        if (node.key.type === AST_NODE_TYPES.Identifier) {
+          methodName = node.key.name;
+        } else {
+          methodNameIsLiteral = true;
+          // null & undefined can be used as property names, undefined counted as Identifier & null as Literal
+          methodName =
+            (node.key as TSESTree.Literal).value?.toString() ?? 'null';
+          methodNameIsNull = (node.key as TSESTree.Literal).value == null;
```

Comment:
> `node.key as TSESTree.Literal`

[Refactor] This `as` assertion isn't always going to be correct. The TSESTree types are set up to represent all the possible values of properties. The fact that an `as` was needed to get this code to type check means that the code isn't handling some case that it needs to handle.

Removing the `as` gives:

```plaintext
Property 'value' does not exist on type 'FunctionExpression | ArrayExpression | ArrayPattern | ArrowFunctionExpression | AssignmentExpression | ... 35 more ... | YieldExpression'.
  Property 'value' does not exist on type 'FunctionExpression'.
```

...which is true. [What if the node's key is a function expression](
https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&showAST=es&code=MYGwhgzhAECSB2A3MICWATAymAtgBxAFNoBvAKGmgG0AzAV3mABdUB7eaACgEpToBfALo9SFStGDsIrIgDoQrAOacARAE1UePIUIBCFdzH8y-IA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA)?

```ts
class InvalidSample {
  [function () { }]() {
    console.log("Yippee!")
  }
}
```

This code is wacky (and causes a TypeScript complaint) but someone in the wild might try to run this rule on that code. 

Change request: remove all the `as` assertions from this file, and explicitly handle those odd node edge cases. 

Note too that we require high unit test coverage (generally, ~100% if possible) for new rule tests. So this'll need unit tests added for any new logic. You can always run `yarn test` from `packages/eslint-plugin` to generate a testing coverage report at `packages/eslint-plugin/coverage/lcov-report/index.html` (or see the Codecov PR comment for an online viewer).